hail/src/main/scala/is/hail/rvd/KeyedRVD.scala

```diff
@@ -72,6 +72,35 @@ class KeyedRVD(val rvd: RVD, val key: Int) {
     }.extendKeyPreservesPartitioning(joinedType.key)
   }
 
+  // 'joinedType.key' must be the join key, followed by the remaining left key,
+  // followed by the (possibly renamed) remaining right key. 'joiner' must copy
+  // these 'joinedType.key' fields from the corresponding fields in the
+  // JoinedRegionValue.
+  def orderedZipJoin(
```

Comment:
There's already an `orderedZipJoin` that returns `ContextRDD` of `JoinedRegionValue`. It's used in concordance, and I couldn't figure out how to make concordance work with a zipJoin like yours. If we do still need the old one, this should probably be moved next to it (one method down), and be implemented as a simple `mapPartitions` on the result of the other one.